src/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -61,32 +62,34 @@ public static Version getVersion(final File zipFile) {
    * If the map cannot be downloaded a message prompt is shown to the user.
    */
   public static void showDownloadMapsWindow(final String mapName) {
-    showDownloadMapsWindow(null, Optional.of(mapName));
-  }
-
-  public static void showDownloadMapsWindow(final Component parent) {
-    showDownloadMapsWindow(parent, Optional.empty());
+    showDownloadMapsWindow(Optional.of(mapName));
   }
 
   public static void showDownloadMapsWindow() {
-    showDownloadMapsWindow(null, Optional.empty());
+    showDownloadMapsWindow(Optional.empty());
   }
 
 
-  private static void showDownloadMapsWindow(final Component parent, Optional<String> mapName) {
-    final DownloadRunnable download = new DownloadRunnable(ClientContext.mapListingSource().getMapListDownloadSite());
+  private static void showDownloadMapsWindow(Optional<String> mapName) {
+    Runnable downloadAndShowWindow = () -> {
```

Comment:
This was an important change, to deal with BAckground task runner no longer blocking.
Basically we set the show download window action to happen at the end of the same thread that does the file list download. Contrast that to doing the file list download, blocking, then in the foreground thread we show the download window.